Backend/Remora.Discord.Caching/Services/MemoryCacheProvider.cs

```diff
@@ -0,0 +1,96 @@
+//
+//  MemoryCacheProvider.cs
+//
+//  Author:
+//       Jarl Gullberg <jarl.gullberg@gmail.com>
+//
+//  Copyright (c) 2017 Jarl Gullberg
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Options;
+using Remora.Results;
+
+namespace Remora.Discord.Caching.Services;
+
+/// <summary>
+/// A <see cref="IMemoryCache"/>-backed cache provider.
+/// </summary>
+public class MemoryCacheProvider : ICacheProvider
+{
+    private readonly IMemoryCache _memoryCache;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MemoryCacheProvider"/> class.
+    /// </summary>
+    /// <param name="memoryCache">The memory cache.</param>
+    public MemoryCacheProvider(IMemoryCache memoryCache)
+    {
+        _memoryCache = memoryCache;
+    }
+
+    /// <inheritdoc cref="ICacheProvider.CacheAsync{TInstance}" />
+    public ValueTask CacheAsync<TInstance>
+        (string key, TInstance instance, TimeSpan? absoluteExpiration = null, TimeSpan? slidingExpiration = null, CancellationToken ct = default)
```

Comment:
Fix formatting here.

Reply:
Would it be more appropriate to put it on one line? Or break it out even more?..

Reply:
The rest of the library uses Allman-style line breaks for very long lines, regardless of where the construct is; that is, this method should put each parameter on its own line, with the parentheses placed like you normally would place braces.